### PR TITLE
feat: klib incremental compilation

### PR DESCRIPTION
**Pull request by danil-pavlov**
* [KT-61242](https://youtrack.jetbrains.com/issue/KT-61242)

### CODE REVIEW

File: docs/topics/native/native-improving-compilation-time.md [hunk 1]
```diff
@@ -72,6 +71,13 @@ Here are some recommendations for configuring Gradle for better compilation perf
   the build completes successfully. It is possible that these properties were added previously to work around issues that
   have already been fixed.
 
+* **Try incremental compilation of klib artifacts**. With incremental compilation, if your change of the project module
+  causes only a partial recompilation of source code into `klib` artifacts, just a part of the `klib` is further
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-improving-compilation-time.md:75]: But it is the other way around: sources are fully recompiled into klibs, but then the compiler attempts to recompile only some parts of klibs into a binary.

File: docs/topics/native/native-improving-compilation-time.md [hunk 2]
```diff
@@ -72,6 +71,13 @@ Here are some recommendations for configuring Gradle for better compilation perf
...
+  recompiled into a binary.
+  
+  This feature is an [Experimental](components-stability.md#stability-levels-explained). To enable it,
+  add the `kotlin.incremental.native=true` option to your `gradle.properties` file.
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-improving-compilation-time.md:79]: Should we explicitly mention that some bugs (e.g. compiler crashes) are possible, and encourage users to report them?

File: docs/topics/native/native-improving-compilation-time.md (hunk 1 above)
Comment (danil-pavlov) [reply docs/topics/native/native-improving-compilation-time.md]: Oh, it was copied from What's new, I suggested a change there as well

**Review by SvyatoslavScherbina — APPROVED** [review]

File: docs/topics/native/native-improving-compilation-time.md [hunk 3]
```diff
@@ -72,6 +71,14 @@ Here are some recommendations for configuring Gradle for better compilation perf
   the build completes successfully. It is possible that these properties were added previously to work around issues that
   have already been fixed.
 
+* **Try incremental compilation of klib artifacts**. With incremental compilation, if only a part of the `klib` artifact
+  produced by the project module changes, just a part of `klib` is further
+  recompiled into a binary.
+
+  This feature is an [Experimental](components-stability.md#stability-levels-explained). To enable it,
```
Comment (sarahhaggarty) [inline docs/topics/native/native-improving-compilation-time.md:78]:
```suggestion
  This feature is [Experimental](components-stability.md#stability-levels-explained). To enable it,
```

File: docs/topics/native/native-improving-compilation-time.md [hunk 4]
```diff
@@ -72,6 +71,14 @@ Here are some recommendations for configuring Gradle for better compilation perf
...
+  add the `kotlin.incremental.native=true` option to your `gradle.properties` file. If you face any issues,
```
Comment (sarahhaggarty) [inline docs/topics/native/native-improving-compilation-time.md:79]:
```suggestion
  add the `kotlin.incremental.native=true` option to your `gradle.properties` file. If you face any problems,
```

**Review by sarahhaggarty — APPROVED** [review]
Looks good! I just have a couple of very minor suggestions. LGTM!

**Pull request merged**